lib/Sema/TypeCheckAttr.cpp

```diff
@@ -7235,6 +7236,15 @@ void AttributeChecker::visitRuntimeMetadataAttr(RuntimeMetadataAttr *attr) {
   }
 }
 
+void AttributeChecker::visitMacroRoleAttr(MacroRoleAttr *attr) {
+  if (attr->getMacroRole() == MacroRole::Declaration &&
+      attr->getNames().empty()) {
+    diagnoseAndRemoveAttr(attr, diag::macro_must_declare_introduced_names,
+                          /*declaration*/0)
+        .fixItInsert(attr->getRParenLoc(), "names: <#names...#>");
+  }
```

Comment:
I don't think this check is correct. A declaration attribute might not introduce any names (like `#warning` or `#error`), or might only introduce declarations with names produced by `createUniqueName()`.

Reply:
I keep forgetting that; fixed!